jetty-core/jetty-client/src/test/java/org/eclipse/jetty/client/HttpClientContentDecoderFactoriesTest.java

```diff
@@ -53,19 +55,28 @@ public boolean handle(Request request, Response response, Callback callback)
         client.getContentDecoderFactories().put(new ContentDecoder.Factory("UPPERCASE")
         {
             @Override
-            public ContentDecoder newContentDecoder()
+            public Content.Source newDecoderContentSource(Content.Source source)
             {
-                return byteBuffer ->
+                return new ContentSourceTransformer(source)
                 {
-                    byte b = byteBuffer.get();
-                    if (b == '*')
-                        return bufferPool.acquire(0, true);
+                    @Override
+                    protected Content.Chunk transform(Content.Chunk chunk)
+                    {
+                        if (chunk.isEmpty())
+                            return chunk.isLast() ? Content.Chunk.EOF : Content.Chunk.EMPTY;
 
-                    RetainableByteBuffer buffer = bufferPool.acquire(1, true);
-                    int pos = BufferUtil.flipToFill(buffer.getByteBuffer());
-                    buffer.getByteBuffer().put(StringUtil.asciiToLowerCase(b));
-                    BufferUtil.flipToFlush(buffer.getByteBuffer(), pos);
-                    return buffer;
+                        ByteBuffer byteBufferIn = chunk.getByteBuffer();
+                        byte b = byteBufferIn.get();
+                        if (b == '*')
+                            return Content.Chunk.EMPTY;
```

Comment:
`return bufferPool.acquire(0, true);`

Reply:
Feels artificial, one would never write an acquire of `0` bytes from the pool.

Reply:
My memory tells me we used to do this because of some unseen bug. This check is cheap and easy, why not keeping it in?